lib/plugins/place_block.js

```diff
@@ -10,7 +10,10 @@ function inject (bot) {
 
     let newBlock = bot.blockAt(dest)
     if (oldBlock.type === newBlock.type) {
-      [oldBlock, newBlock] = await onceWithCleanup(bot, `blockUpdate:${dest}`, { timeout: 5000 })
+      [oldBlock, newBlock] = await onceWithCleanup(bot, `blockUpdate:${dest}`, {
+        timeout: 5000,
+        checkCondition: (oldBlock, newBlock) => !oldBlock || !newBlock || oldBlock.type !== newBlock.type
```

Comment:
can you add a comment to explain this condition? it's not obvious to me why we want this

Reply:
When testing 1.20, I kept getting block updates of the same block type before the block update saying the block had changed from the server making the function think the place had failed.

Reply:
Rereading this and I still don't get it
The condition is :
* When there is no old block
* or when there no new block
* or when the new block is different from the old

Can you explain why the first 2 are wanted ?

For example why is a change from null to null ok ?

Reply:
Because of this
![image](https://github.com/PrismarineJS/mineflayer/assets/66561610/fdc68f03-8779-46f5-935d-b42f362ee0ae)

Reply:
no that's different

this if is triggering only when both oldBlock and newBlock are null

Reply:
I think I just did that because if only one of oldBlock and newBlock is null the third check will be true anyway so easier to just write them all as or.